### PR TITLE
Fix cargo clippy

### PR DESCRIPTION
**Pull request by zhiburt**
Hi there

I was kind of studying the source code and decided to delivery a clippy fixes along the way.

A note:

Unit address_size is unused so could be deleted.
Maybe there's some plan for it so I've left it.

PS: if you don't mind could you elaborate on this comment?

```
// TODO: handle `lea rax, [rip + offset]; call rax`
```

As I understand the offset will be no calculated correctly in case of lea usage?

### CODE REVIEW

**Comment by zhiburt** [comment]
Please don't merge it.

I've incidentally found that now there a bunch of

```
Segmentation fault (core dumped)
```

File: parser/src/namespace.rs [hunk 1]
```diff
@@ -35,7 +35,7 @@ impl<'input> Namespace<'input> {
 
     /// The parent namespace.
     pub fn parent(&self) -> Option<&Namespace<'input>> {
-        self.parent.as_ref().map(|x| &**x)
```
Comment (zhiburt) [inline parser/src/namespace.rs:38]: I think it's because of this change.

Which actually implies that it is a false-positive in clippy?

Comment (zhiburt) [reply parser/src/namespace.rs:38]: Not true still

Segmentation fault

**Comment by zhiburt** [comment]
OK it was a `mem::transumute` as I understand.
Not sure exactly why it happend though.

**Comment by philipc** [comment]
Thanks for working on this. However, I find some of the clippy warnings to be too opinionated, and I prefer not to apply them. I do occasionally go through and fix warnings I agree with, and I think it would be a better use of my time to do that myself, rather than review this PR and try to do a back and forth to get it into a state that I'm happy to merge.

In general, it's not a good idea to fix warnings (either compiler warnings or clippy warnings) without understanding the code that you are changing. This especially applies to `unsafe` code.

> PS: if you don't mind could you elaborate on this comment?
> 
> ```
> // TODO: handle `lea rax, [rip + offset]; call rax`
> ```
> 
> As I understand the offset will be no calculated correctly in case of lea usage?

The purpose of this code is to find all addresses that the function calls. However, we currently only look at individual `call` instructions. Thus if the individual instruction is `call rax`, we currently can't determine what address that is calling, but in theory we should be able to work out what this address is in some cases, such as if the previous instruction was a RIP relative offset calculation.

**Pull request closed without merging**